From d5c628a6179b99705fa05ab87437321b132c81bc Mon Sep 17 00:00:00 2001
From: Brendan Heading <brendanheading@gmail.com>
Date: Wed, 21 Jun 2017 16:36:15 -0400
Subject: [PATCH] Fix compile time atomic detection

Improved compile-time detection of atomic support in the compiler.

Upstream-Status: 'Needs information'
See : https://gitlab.gnome.org/GNOME/glib/issues/1063

Signed-off-by: Brendan Heading <brendanheading@gmail.com>
Signed-off-by: Adam Duskett <aduskett@gmail.com>
Refresh for 2.68.1
---
 glib/gthread-posix.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/glib/gthread-posix.c b/glib/gthread-posix.c
index 71a6b7b..d09d029 100644
--- a/glib/gthread-posix.c
+++ b/glib/gthread-posix.c
@@ -72,8 +72,7 @@
 #include <sys/syscall.h>
 #endif
 
-#if defined(HAVE_FUTEX) && \
-    (defined(HAVE_STDATOMIC_H) || defined(__ATOMIC_SEQ_CST))
+#if defined(HAVE_FUTEX) && defined(__ATOMIC_SEQ_CST) && defined(__GCC_HAVE_SYNC_COMPARE_AND_SWAP_4) && !defined(__clang__)
 #define USE_NATIVE_MUTEX
 #endif
 
-- 
2.9.4

